cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -50,6 +50,11 @@ namespace compute {
 SCALAR_ARITHMETIC_UNARY(AbsoluteValue, "abs", "abs_checked")
 SCALAR_ARITHMETIC_UNARY(Negate, "negate", "negate_checked")
 
+Result<Datum> Sign(const Datum& arg, SignOptions options, ExecContext* ctx) {
+  auto func_name = options.with_signed_zero ? "sign_with_signed_zero" : "sign";
```

Comment:
I'm still not sure why we need both variants. Is this a user request? @ianmcook ?

Reply:
Note that we could expose a `copysign` instead of `sign_with_signed_zero`, which would be more generic.
(`sign_with_signed_zero(x)` would simply be `copysign(x, 1)`, AFAICT)

Reply:
@pitrou The sign kernel is to be used by the rounding function. I agree with the fact that `sign_with_signed_zero` is equivalent to `copysign`. Actually, `copysign` is the variant needed for rounding function.

Nevertheless, `sign` is a common function exposed in frontend tools, so I consider it worthwhile to provide.

Reply:
Also, I consider that `copysign` should not be exposed as a compute function but rather used internally, and instead we should expose `sign` and `sign_with_signed_zero` (implementations may use `copysign(x, 1)`. The rationale for this is that `copysign(x, 1)` is not common in any SQL, dataframes library but `sign` is.

A use case of `sign` is to build a custom ranking based on the signedness of values.
For example, consider ranking values relative to a constant value (e.g., 3):
```
SELECT CASE 
    WHEN @a < 3 THEN 0
    WHEN @a = 3 THEN 1
    WHEN @a > 3 THEN 2
END
```
is not supported because expressions are not allowed in `CASE-WHEN` constructs, so a workaround is
```
SELECT CASE SIGN(@a - 3)
    WHEN -1 THEN 0
    WHEN 0 THEN 1
    WHEN 1 THEN 2
END
```

Reply:
Please, can we answer the original question? Why are *two* variants needed?

Reply:
After careful thought, there is not a general need for having two variants. The only case where I see that `sign_with_signed_zero` can be relevant is if data stored is used for calculus-related operations (e.g., integral limits).
I will remove the `sign_with_signed_zero` variant.